main.go

```diff
@@ -225,6 +232,15 @@ func main() {
 	if os.Getenv("ENABLE_WEBHOOKS") != "false" {
 		setupLog.Info("setting up webhooks")
 		setUpWebhooks(mgr)
+		server := mgr.GetWebhookServer()
```

Comment:
Should this go in the setUpWebhooks function?

Reply:
Actually, on further thought, does this currently do anything? I don’t see `server` used or passed in anywhere?

Reply:
I can add it to the setup function

Reply:
I have printed some logs to ensure that the tls options for the webhook server is persisted. After appending, I fetch again and print the values again:

<img width="1039" alt="Screenshot 2023-10-26 at 5 02 15 PM" src="https://github.com/redhat-appstudio/application-service/assets/31771087/c4d7f60f-37fc-48c1-943f-6748951984c2">


```
{"level":"info","ts":"2023-10-26T20:59:26.464Z","logger":"setup","msg":"disabling http/2 on the webhook server","controllerKind":""}
{"level":"info","ts":"2023-10-26T20:59:26.464Z","logger":"setup","msg":"getting webhook server tlsopt before appending []","controllerKind":""}
{"level":"info","ts":"2023-10-26T20:59:26.464Z","logger":"setup","msg":"getting webhook server tlsopt after appending [0x1bc49a0]","controllerKind":""}
{"level":"info","ts":"2023-10-26T20:59:26.464Z","logger":"setup","msg":"webhook server tlsopt config value after appending [http/1.1]","controllerKind":""}
{"level":"info","ts":"2023-10-26T20:59:26.464Z","logger":"setup","msg":"getting the webhook server again to make sure tlsopt is persisted: [0x1bc49a0]","controllerKind":""}
{"level":"info","ts":"2023-10-26T20:59:26.464Z","logger":"setup","msg":"webhook server tlsopt config value after getting again [http/1.1]","controllerKind":""}
{"level":"info","ts":"2023-10-26T20:59:26.464Z","logger":"setup","msg":"starting manager","controllerKind":""}
{"level":"info","ts":"2023-10-26T20:59:26.464Z","logger":"controller-runtime.webhook.webhooks","msg":"Starting webhook server"}

```

Reply:
Right, but that's just changing the options, and re-getting the options struct. Do we know if that actually gets reflected in the server without needing to restart it though?

Reply:
I can't remember what port the webhook service runs on, but maybe you could `oc port-forward` the webhook service to localhost, and checking the headers?

Reply:
updated!

Reply:
Let me take a look at the headers.

Reply:
Hmm it seems regardless of setting it to http/1.1, the protocl is always http/1:

```
MacBook-Pro:samples maysun$ curl -I localhost:9443
HTTP/1.0 400 Bad Request
```

Reply:
After a little more digging it seems that our control-runtime v0.14.6 does not support http/2, where as controller-runtime v0.16.3 supports http2 by default

Reply:
Something to probably check when we upgrade controller runtime versions

Reply:
My above statement was incorrect and i have since striked it out.

On a 400 Bad Request, the server defaults to HTTP/1.0 because that is how the controller-runtime sets it [here](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.14.6/pkg/webhook/server.go#L241)

Curling to a good endpoint
- with http/1.1 enabled (http/2 removed)
```
MacBook-Pro:samples maysun$ curl -k -I  --header "Content-Type: application/json" --request POST https://localhost:9443/mutate-appstudio-redhat-com-v1alpha1-component
HTTP/1.1 200 OK
Date: Fri, 27 Oct 2023 20:37:21 GMT
Content-Length: 178
Content-Type: text/plain; charset=utf-8
```
- with http/2 enabled
```
MacBook-Pro:samples maysun$ curl -k -I  --header "Content-Type: application/json" --request POST https://localhost:9443/mutate-appstudio-redhat-com-v1alpha1-component
HTTP/2 200 
content-type: text/plain; charset=utf-8
content-length: 178
date: Fri, 27 Oct 2023 20:32:23 GMT
```